### PR TITLE
Feature/home screen

### PR DESCRIPTION
**Pull request by armaan-n**
TLDR; The home screen transition form view to view is working, but there are issues with passing state from one use case to another, which must be addressed in a future PR.

# Requirements : 

The user should be able to navigate from the home screen to the other use cases via buttons. The user should also be able to log out, which will return them to the sign up screen. When going from the home screen to another use case, they should be able to perform actions relevant to their own profile, have these changes saved with respect to their own profile, and be able to see these changes reflected if they log out and log back in. Also, immediately after creating a request, the user should see a map displaying the route from the doctor to the patient. Navigating away from and back to the home screen should remove the map.

# Implementation Details :

The use case requires no data access object.

Most of the logic has to do with state transition. A switch case in the presenter will determine which screen is transitioned to. The data eventually received by the presenter is determined by which button is pressed by the user, and the corresponding logic can be found in the view.

I've also implemented the main method, so our app can now properly display the GUI to the user. 

I created a small helper use case call return to lock, which allows to user to navigate from the home screen, sign up screen and log in screen back to the initial lock screen.

A lot of refactoring had to be done. Most notably, I wrote factories for the view request, lock, and edited use cases, and rewrote to create, edit, login and signup use case factories to accommodate unforeseen dependencies (return to home screen use case). For the same reason, the views had to be modified.

Data access objects were modified to remove redundancies and implement use case interfaces that hadn't already been implemented.

Lock was also missing its interface adapters, so I implemented those as well.

# Testing :

No testing has been done. The purpose of this PR is to pull a minimum viable product into the develop branch. More testing will be done in a future PR.

# Notes :

The functionality is not all there. I will create a bugfix branch to address some of the issues, namely, the inability to create requests, the inability to edit one's profile, and the inability to sign up.

### CODE REVIEW

**Comment by michellengnx** [comment]
bro down bad

**Comment by ISMAELCHONA04** [comment]
Very informative Java doc. Try to make more frequent commits. Code is satisfictory

**Review by ISMAELCHONA04 — CHANGES_REQUESTED** [review]
The tests are not passing however the app creates views and it works well. Signup and log in worked properly. Edit profile should have all inputs prepopulated to be edited but they are empty at the moment.

**Review by izorakayumova — COMMENTED** [review]
Went to your branch and the app looks amazing right now! Your commits are very informative and I can help add javadoc to the files missing them. When running the app to check correctness, I noticed that editprofile is not prepopulated since you have changed the logic to not change the functions if the text fields are empty - I will edit my PR requests.
